Mathlib/Data/Option/Basic.lean

```diff
@@ -68,6 +68,11 @@ theorem eq_of_mem_of_mem {a : α} {o1 o2 : Option α} (h1 : a ∈ o1) (h2 : a 
   h1.trans h2.symm
 #align option.eq_of_mem_of_mem Option.eq_of_mem_of_mem
 
+-- TODO: Is this lemma worth having? If not, it can just be
+-- replaced with `(by simp)` where it's used.
+theorem mem_some_eucl {a a' b : α} : b ∈ some a → b ∈ some a' → a = a' :=
```

Comment:
It certainly needs renaming, I can't work out what `eucl` is.

`eq_of_mem_some_of_mem_some`?

Reply:
Please remove.

Reply:
I think this should be inlined into the statement of the other lemma

Reply:
Removed! My thought with naming was that this is a proof that `(· ∈ some ·)` is a Euclidean relation.